### ISSUE TITLE
desi_compute_sky singular matrix

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#1470]
With the latest daily rerun, we're getting a hundreds of thousands of "Singular matrix" warnings from desi_compute_sky, when previously they were rare (0-2 per frame).

Example:
```
desi_compute_sky -i /global/cfs/cdirs/desi/spectro/redux/daily/exposures/20211016/00104546/frame-r8-00104546.fits --fiberflat /global/cfs/cdirs/desi/spectro/redux/daily/calibnight/20211016/fiberflatnight-r8-20211016.fits -o testsky.fits --adjust-wavelength --adjust-lsf --pca-corr /global/cfs/cdirs/desi/spectro/desi_spectro_calib/trunk/spec/sm2/skycorr-pca-sm2-r8.fits
```

In some cases this was fatal (e.g. r4 and r8 didn't get sky modesl for that exposure), while for others like 20211016/00104561 it generated >100k warning messages but still wrote output, vs. the previous daily run which had 0 Singular matrix warnings.  (previous = when that daily was first run, not the one that we ran over the weekend and then deleted).

Note that the original inputs are available under daily/attic/rerunarchive-20211029/exposures... to test if the problem is with some upstream issue like masking, or something about the latest sky model code.

**Comment by julienguy** [comment]
This is due to the fact an amplifier is masked such that the inverse variance = 0  for many sky lines causing the fit of the line center and width to fail.
This is not by itself a serious issue (except we can improve the selection of valid fibers upstream and provide a more useful warning message) but it exposes a more serious problem: the FIBERSTATUS bit fibermask.BADREADNOISE is not set for the fiber traces on the CCD quadrant with ccdmask.BADREADNOISE. We have to change this.

**Comment by sbailey** [comment]
Fixed in PR #1472; closing ticket.
